File: 169-MajorityElement/MajorityElement.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums_count = {}
        for num in nums:
            nums_count[num] = nums_count.get(num, 0) + 1
        return max(nums_count.keys(), key = lambda k: nums_count[k])

File: 169-MajorityElement/MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> counter;
        for (const auto& num : nums) {
            counter[num]++;
        }
        int max_key = -1;
        int max_value = -1;
        for (const auto& [key, value] : counter) {
            if (value > max_value) {
                max_key = key;
                max_value = value;
            }
        }
        return max_key;
    }
};